src/components/FilterAttachedResults/FilterAttachedResults.ts

```diff
@@ -0,0 +1,149 @@
+
+        this.options = ComponentOptions.initComponentOptions(
+            element,
+            FilterAttachedResults,
+            options
+        );
+
+        this.initialize();
+
+        this.queryStateModel.registerNewAttribute(
+            QueryStateModel.getFacetId(FilterAttachedResults.ID),
+            false
+        );
+        this.bind.onRootElement(
+            QueryEvents.buildingQuery,
+            (args: IBuildingQueryEventArgs) => this.handleBuildingQuery(args)
+        );
+        this.bind.onQueryState(
+            'change:',
+            QueryStateModel.getFacetId(FilterAttachedResults.ID),
+            (args: IAttributesChangedEventArg) =>
+                this.handleQueryStateChange(args)
+        );
+    }
+
+    public isSelected(): boolean {
+        return this.checkbox && this.checkbox.isSelected();
+    }
+
+    public toggle(): void {
+        if (this.isSelected()) {
+            this.checkbox.reset();
+        } else {
+            this.checkbox.select(true);
+        }
+    }
+
+    protected initialize(): void {
+        let mainSection = $$('div', { className: 'CoveoFacet' });
+        let headerSection = $$('div', { className: 'coveo-facet-header' });
+        const labelDiv = $$('label', {
+            className: 'coveo-facet-value-label-wrapper'
+        }).el;
+        headerSection.append(labelDiv);
+        mainSection.append(headerSection.el);
+
+        this.createCheckbox().then(checkbox => {
+            this.checkbox = checkbox;
+            labelDiv.appendChild(this.checkbox.getElement());
+        });
+
+        this.element.appendChild(mainSection.el);
+    }
+
+    private async createCheckbox(): Promise<Checkbox> {
+        if (Coveo.Checkbox === undefined) {
+            await load('Checkbox');
+        }
+        return new Checkbox(
+            checkbox => this.handleCheckboxChange(),
+            this.options.text
+        );
+    }
+
+    private handleCheckboxChange() {
+        this.queryStateModel.set(
+            QueryStateModel.getFacetId(FilterAttachedResults.ID),
+            this.checkbox.isSelected()
+        );
+        this.triggerQuery();
+    }
+
+    private triggerQuery() {
+        this.usageAnalytics.logSearchEvent<IAnalyticsAttachedResultsFilterMeta>(
+            { name: 'filterAttachedResults', type: 'misc' },
+            { filteredAttachedResults: this.checkbox.isSelected() }
+        );
+        this.queryController.executeQuery({ origin: this });
+    }
+
+    private handleQueryStateChange(args: IAttributesChangedEventArg) {
+        if ((args as any)['value']) {
+            this.checkbox.select(false);
+        } else {
+            this.checkbox.reset();
+        }
+    }
+
+    private handleBuildingQuery(args: IBuildingQueryEventArgs) {
+        if (this.checkbox.isSelected()) {
+            const uriHashes = this.options.getUriHashes();
+            args.queryBuilder.advancedExpression.add(
+                '@urihash=(' + uriHashes.join(',') + ')'
```

Comment:
Could probably check if `uriHashes` is empty before adding it to the advancedExpression?

```JavaScript
var uriHashes=[]; 
'@urihash=(' + uriHashes.join(',') + ')'
// "@urihash=()"
```

Reply:
not really, if there is nothing to show then we want to show nothing

Reply:
I'm saying, if uriHashes is empty, the query will contain `@urihash=()`.

You should check first if `uriHashes` is empty, and simply don't add anything to the query if it's empty?

Reply:
No that's what i'm saying, if there is notrhing to show we still want to add it to the query so nothing will show in the results, which is correct !